test/controllers/incomeSources/add/AddBusinessTradeControllerSpec.scala

```diff
@@ -83,340 +84,231 @@ class AddBusinessTradeControllerSpec extends TestSupport
       ec = ec
     )
 
-  "AddBusinessTradeController" should {
-    "redirect a user back to the custom error page" when {
-      "the individual is not authenticated" should {
-        "redirect them to sign in" in {
-          setupMockAuthorisationException()
-          val result = TestAddBusinessTradeController.show(isAgent = false, isChange = false)(fakeRequestWithActiveSession)
-          status(result) shouldBe SEE_OTHER
-          redirectLocation(result) shouldBe Some(controllers.routes.SignInController.signIn.url)
-        }
-      }
-    }
-    "the user has timed out" should {
-      "redirect to the session timeout page" in {
-        setupMockAuthorisationException()
-
-        val result = TestAddBusinessTradeController.submit(isAgent = false, isChange = false)(fakeRequestWithTimeoutSession)
-
-        status(result) shouldBe SEE_OTHER
-        redirectLocation(result) shouldBe Some(controllers.timeout.routes.SessionTimeoutController.timeout.url)
-      }
-    }
-
-    ".show" should {
-      "show correct page when user valid" in {
-        disableAllSwitches()
-        enable(IncomeSources)
+  def authenticate(isAgent: Boolean): Unit = {
+    if (isAgent) setupMockAgentAuthRetrievalSuccess(testAgentAuthRetrievalSuccess)
+    else setupMockAuthRetrievalSuccess(testIndividualAuthSuccessWithSaUtrResponse())
+  }
 
-        setupMockAuthRetrievalSuccess(BaseTestConstants.testIndividualAuthSuccessWithSaUtrResponse())
-        setupMockGetIncomeSourceDetails()(businessesAndPropertyIncome)
-        setupMockCreateSession(true)
-        setupMockGetMongo(Right(Some(sessionDataName)))
+  def getRequest(isAgent: Boolean) = {
+    if (isAgent) fakeRequestConfirmedClient()
+    else fakeRequestWithActiveSession
+  }
+  def postRequest(isAgent: Boolean) = {
+    if (isAgent) fakePostRequestConfirmedClient()
+    else fakePostRequestWithActiveSession
+  }
 
-        val result: Future[Result] = TestAddBusinessTradeController.show(isAgent = false, isChange = false)(fakeRequestWithActiveSession)
-        status(result) shouldBe OK
+  for (isAgent <- Seq(true, false)) yield {
+    s"AddBusinessTradeController ${if (isAgent) "Agent" else "Individual"}" should {
```

Comment:
💭 Change to AddBusinessTradeController.show in matcher to make more clear perhaps